presto-parquet/src/main/java/com/facebook/presto/parquet/reader/MetadataReader.java

```diff
@@ -85,11 +86,11 @@ public static ParquetMetadata readFooter(FSDataInputStream inputStream, Path fil
         validateParquet(fileSize >= MAGIC.length + PARQUET_METADATA_LENGTH + MAGIC.length, "%s is not a valid Parquet File", file);
         long metadataLengthIndex = fileSize - PARQUET_METADATA_LENGTH - MAGIC.length;
 
-        inputStream.seek(metadataLengthIndex);
-        int metadataLength = readIntLittleEndian(inputStream);
+        InputStream tailStream = readFully(inputStream, metadataLengthIndex, PARQUET_METADATA_LENGTH + MAGIC.length);
+        int metadataLength = readIntLittleEndian(tailStream);
 
         byte[] magic = new byte[MAGIC.length];
-        inputStream.readFully(magic);
+        validateParquet(tailStream.read(magic) == MAGIC.length, "Not valid Parquet file: no enough data for tailing MAGIC bytes");
```

Comment:
is this needed? the following line is validating all bytes of magic
could tailStream.read(magic) enough?

Reply:
Changed to `tailStream.read(magic)`. Thanks!